transport/controlbuf.go

```diff
@@ -357,10 +357,9 @@ func newLoopyWriter(s side, fr *framer, cbuf *controlBuffer, bdpEst *bdpEstimato
 const minBatchSize = 1000
 
 // run should be run in a separate goroutine.
-func (l *loopyWriter) run() {
+func (l *loopyWriter) run() (err error) {
```

Comment:
Even though `err` is a named return variable, please don't use bare `return`s to avoid potential problems.  You can then remove these explicit variable declarations entirely by using `:=` where `it` and `isEmpty` are assigned below, which is usually worth doing.

Reply:
Updated the return statements but can't using `:=` for vars since there's a logging call in defer that needs `err`.